python-checks/src/test/resources/checks/confusingTypeChecking/nonCallableCalled.py

```diff
@@ -4,32 +4,38 @@
 def empty_union(x: Union['A', 'B']):
   x()
 
-def foo(param1: int, param2: Set[int], param3: FrozenSet[int]):
+def foo(param1: int, param2: Set[int], param3: FrozenSet[int], param4: list[str], param5: set[int]):
   param1() # Noncompliant {{Fix this call; Previous type checks suggest that "param1" has type int and it is not callable.}}
   x = 42
   x() # OK, raised by S5756
 
-  param2() # Noncompliant
+  param2() # FN SONARPY-2022
   s = set()
   s() # OK, raised by S5756
 
-  param3() # Noncompliant
+  param3() # FN SONARPY-2022
   fs = frozenset()
   fs() # OK, raised by S5756
 
+  for item in param4:
+      item() # Noncompliant
+
+  param5() # Noncompliant
+
 def derived(x: int):
-  x.conjugate()() # NonCompliant
+  x.conjugate()() # FN need return type's type source resolution SONARPY-2024
   y = x or 'str'
-  y() # Noncompliant
+  y() # FN Value calculation with involved parameters support is needed (SONARPY-2023)
   z = x + 42
-  z() # Noncompliant
+  z() # FN Value calculation with involved parameters support is needed (SONARPY-2023)
 
 class Base: ...
 class CallableBase:
   def __call__(self, *args, **kwargs): ...
 def with_isinstance(x: Base):
   if isinstance(x, CallableBase):
-    x()
+    # FP type check guard need to be supported in V2 type inference (SONARPY-2017)
```

Comment:
As discussed, I'd probably like to see some of the FP/FNs tickets addressed before merging the PR for the rule itself (if only to know which ones we can actually tackle simply). Open to discussion though.